interfaces/cython/cantera/composite.py

```diff
@@ -590,7 +596,7 @@ def append(self, state=None, **kwargs):
             raise IndexError("Can only append to 1D SolutionArray")
 
         for name, value in self._extra.items():
-            value.append(kwargs.pop(name))
+            np.append(value,kwargs.pop(name))
```

Comment:
```suggestion
            np.append(value, kwargs.pop(name))
```

Reply:
But this isn't correct, `np.append` does not work in-place. See the documentation: https://numpy.org/doc/1.18/reference/generated/numpy.append.html This will most likely be very slow, since NumPy allocates a new array every time `np.append()` is called. I wonder if it would be better to cast the values in `_extra` to lists and use the list append method, and then cast them back to arrays? Maybe the complexity isn't worth it though.

Reply:
I also can't find any other way around [see](https://stackoverflow.com/questions/13215525/how-to-extend-an-array-in-place-in-numpy)

Reply:
What do you mean, any other way around? You'll have to find some way, since this won't work as expected. I think the simplest method is just to reassign `value`: `value = np.append(value, kwargs.pop(name))` That might be slow, since NumPy has to create a new array every time, but hopefully not *too* slow.